lib/parse_date/int_from_string.rb

```diff
@@ -53,32 +42,15 @@ def self.latest_year(date_str)
       return if date_str == '0000-00-00' # shpc collection has these useless dates
 
       # B.C. first (match longest string first)
-      return ParseDate.send(:last_year_mult_centuries_bc, date_str) if date_str.match(YY_YY_CENTURY_BC_REGEX)
-      return ParseDate.send(:between_bc_latest_year, date_str) if date_str.match(BETWEEN_Yn_AND_Yn_BC_REGEX)
-      return ParseDate.send(:last_year_for_bc_century, date_str) if date_str.match(BC_CENTURY_REGEX)
-      return ParseDate.send(:year_int_for_bc, date_str) if date_str.match(BC_REGEX)
+      bc_result = ParseDate.send(:latest_year_bc_parsing, date_str)
+      return bc_result if bc_result
 
-      result ||= ParseDate.send(:between_latest_year, date_str)
-      result ||= ParseDate.send(:hyphen_4digit_latest_year, date_str)
-      result ||= ParseDate.send(:hyphen_2digit_latest_year, date_str)
-      result ||= ParseDate.send(:hyphen_1digit_latest_year, date_str)
-      result ||= ParseDate.send(:yyuu_after_hyphen, date_str)
-      result ||= ParseDate.send(:year_after_or, date_str)
-      result ||= ParseDate.send(:negative_4digits_after_hyphen, date_str)
-      result ||= ParseDate.send(:negative_first_four_digits, date_str)
-      result ||= ParseDate.send(:last_year_for_0s_decade, date_str)
-      result ||= ParseDate.send(:first_four_digits, date_str)
-      result ||= ParseDate.send(:year_from_mm_dd_yy, date_str)
-      result ||= ParseDate.send(:last_year_for_decade, date_str) # 198x or 201x
-      result ||= ParseDate.send(:last_year_mult_centuries, date_str) # nth-nth century
-      result ||= ParseDate.send(:last_year_for_century, date_str)
-      result ||= ParseDate.send(:last_year_for_early_numeric, date_str)
```

Comment:
this is the especially annoying one because every new parsing pattern added to the length here.